src/ds3.c

```diff
@@ -2524,6 +2455,83 @@ ds3_error* ds3_get_available_chunks(const ds3_client* client, const ds3_request*
     return NULL;
 }
 
+static ds3_error* _parse_jobs_list(const ds3_log* log, xmlDocPtr doc, ds3_get_jobs_response** _response){
+    ds3_get_jobs_response* response;
+    xmlNodePtr root, child_node;
+    GPtrArray* jobs_array = g_ptr_array_new();
+    ds3_bulk_response* job;
+    ds3_error* error;
+
+    root = xmlDocGetRootElement(doc);
+
+    if(element_equal(root, "Jobs") == false) {
+        char* message = g_strconcat("Expected the root element to be 'Jobs'.  The actual response is: ", root->name, NULL);
+        xmlFreeDoc(doc);
+        ds3_error* error = _ds3_create_error(DS3_ERROR_INVALID_XML, message);
+        g_free(message);
+        return error;
+    }
+
+    response = g_new0(ds3_get_jobs_response, 1);
+
+    for(child_node = root->xmlChildrenNode; child_node != NULL; child_node = child_node->next) {
+        if(element_equal(child_node, "Job") == true) {
+            job = g_new0(ds3_bulk_response, 1);
+            error = _parse_bulk_response_attributes(log, doc, child_node, job);
+            if( error )
+            {
+              LOG(log, DS3_ERROR, "Error parsing bulk_response element");
```

Comment:
If there is an error here, why are we still appending `job`?  It seems like we should free `job` and then not append it.